tests/integ/modin/groupby/test_pct_change.py

```diff
@@ -0,0 +1,132 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+import modin.pandas as pd
+import pandas as native_pd
+import pytest
+
+from snowflake.snowpark.exceptions import SnowparkSQLException
+from tests.integ.modin.groupby.conftest import multiindex_data
+from tests.integ.modin.utils import (
+    create_test_dfs,
+    create_test_series,
+    eval_snowpark_pandas_result,
+)
+from tests.integ.utils.sql_counter import sql_count_checker
+
+BASIC_DATA = [
+    ["a", 1, 1.1],
+    ["a", 1, 1.2],
+    ["a", 2, 1.3],
+    ["b", 2, 1.4],
+    ["a", 3, 1.5],
+    ["b", 4, 1.6],
+    ["c", 1, 0.9],
+]
+
+
+@pytest.mark.parametrize("periods", [0, 1, 2, -1])
```

Comment:
We should get some tests with NA values to make sure they are propagated (or filled correctly if fill_method is specified).

Reply:
Added! (also caught some bugs with it)